### PR TITLE
Add OneHotEncoder FeatureOperation

### LINKED ISSUE #52: FeatureOperation: OneHotEncoder
[3c ii]

### PR DESCRIPTION
**Pull request by alessiamarcolini**
fixes #52

### CODE REVIEW

**Comment by alessiamarcolini** [comment]
fixes #99

**Comment by lorenz-gorini** [comment]
Sorry. I still need to take a look. After today's meeting, I should be able to do that. I also remembered that if a FeatureOperation has all original columns (`columns` attribute) that are in `metadata_cols`, it may be reasonable to automatically include the `derived_columns` in the `metadata_cols` attribute of `Dataset` (columns derived from metadata columns can be considered metadata columns as well, can't they?) . We could discuss about this possibility (this was mostly a reminder for me) :D

**Comment by alessiamarcolini** [comment]
> that if a FeatureOperation has all original columns (columns attribute) that are in metadata_cols, it may be reasonable to automatically include the derived_columns in the metadata_cols attribute of Dataset

we are already doing this here: https://github.com/HK3-Lab-Team/pytrousse/blob/master/src/trousse/dataset.py#L638 
😉

**Comment by lorenz-gorini** [comment]
> > that if a FeatureOperation has all original columns (columns attribute) that are in metadata_cols, it may be reasonable to automatically include the derived_columns in the metadata_cols attribute of Dataset
> 
> we are already doing this here: https://github.com/HK3-Lab-Team/pytrousse/blob/master/src/trousse/dataset.py#L638
> wink

Great. One less thing to discuss and do. Only 2999 left!

**Review by lorenz-gorini — APPROVED** [review]
Great job!

File: src/trousse/feature_operations.py [hunk 1]
```diff
@@ -459,6 +461,242 @@ def __str__(self) -> str:
...
+        dataset : Dataset
+            The dataset to apply the operation on
+
+        Returns
+        -------
+        Dataset
+            New Dataset instance with the operation applied on
+        """
+        dataset = copy.deepcopy(dataset)
+        data = dataset.data[[self.columns[0]]]
```
[the remark this answers is not in the document]
Comment (alessiamarcolini) [reply src/trousse/feature_operations.py:519]: If you look at the code used previously to perform the encoding, you will notice that the input of the `fit` method of the sklearn encoder was a 2d NumPy array (REF: https://github.com/HK3-Lab-Team/pytrousse/blob/master/src/trousse/feature_fix.py#L201). 

In fact, it expects `X: array-like, shape [n_samples, n_features]` (REF: https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html). That is why we cannot use a Series (and I decided to use a DataFrame)

**Pull request merged**